execute/exectypes/outcome.go

```diff
@@ -7,8 +7,40 @@ import (
 	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
 )
 
+type PluginState int
+
+const (
+	// Unknown is the default value indicating there was no previous state.
+	Unknown PluginState = iota
+	GetCommitReports
+	GetMessages
+	Filter
+)
+
+func (p PluginState) Next() PluginState {
+	switch p {
+	case GetCommitReports:
+		return GetMessages
+
+	case GetMessages:
+		// TODO: go to Filter after GetMessages
+		return GetCommitReports
+
+	case Unknown:
```

Comment:
`// Unknown is the default value indicating there was no previous state.`

If there was no previous state, should it start getting new messages? Are we going to start the plugin state with Unknown? If we expect to start with it, why fallthrough?

Reply:
Basically, I chose to do it this way so that an uninitialized previous outcome (i.e. the first round after startup) can transition to the "Next" state with no special handling. So you can get to the first state from `Unknown` or `Filter` (or `GetMessages` in this PR, since I haven't implemented `Filter` yet).